library/ajax/sql_server_status.php

```diff
@@ -20,25 +20,33 @@
 
 require_once(__DIR__ . '/../../interface/globals.php');
 
+use OpenEMR\Common\Csrf\CsrfUtils;
+
+if (!empty($_POST)) {
+    if (!CsrfUtils::verifyCsrfToken($_POST["csrf_token_form"])) {
+        CsrfUtils::csrfNotVerified();
+    }
+}
+
```

Comment:
Bad actor can still attack this by running sql_upgrade, scraping csrf from it, and then running this script continually to capture random queries that are run on the server. This makes it harder but the vulnerability is still here. Even a auth user having access to this could be problematic (bad actor receptionist just collecting queries all day). Been putting thought into this and can't figure out a way to secure this thing. Again, late at night, though, so maybe just need some daylight :)

Reply:
If we could just tie this query to only ever be called by sql upgrade logic within the sql upgrade functions.

Reply:
Maybe a invisible iframe in sql-upgrade script that runs a script with these queries and stores them in temp place in sql column or drive that the polling script then collects from. Maybe option or would that also be vulnerable?

Reply:
Yep, I think that would also be vulnerable. Like an impossible game of chicken or the egg. brain hurts.
@sjpadgett , will need to await your fine mind to solve this dilemma :)

Reply:
I put csrf in to lock down the sql query script. I'm only interested in activity so i'm going to only return the query command, ie alter,update, select etc.. 
If somebody wants to watch that scroll by all day, cool...

Reply:
sounds good. knew your fine mind would sort it out :)

Reply:
yeah right, about as fine as wine in a carton!:) I'm prob fix this then bring in. I've got to quit these diversion and concentrate on onc...

Reply:
finer than my mind. i was gonna suggest just having the script return random uplifting quotes : https://www.davidmarotz.com/quotes.php
:)